flink-runtime/src/main/java/org/apache/flink/runtime/metrics/ReporterSetup.java

```diff
@@ -464,6 +466,21 @@ private static Optional<MetricReporter> loadViaReflection(
                     alternativeFactoryClassName, reporterName, reporterConfig, reporterFactories);
         }
 
-        return Optional.of((MetricReporter) reporterClass.newInstance());
+        try {
+            return Optional.of((MetricReporter) reporterClass.newInstance());
+        } catch (InstantiationException e) {
+            if (ExceptionUtils.findThrowable(e, NoSuchMethodException.class).isPresent()) {
+                throw new IllegalConfigurationException(
+                        String.format(
+                                "%s does not implement a default constructor but is explicitly configured. "
+                                        + "Reflection-based instantiation of reporter instances is deprecated "
+                                        + "and does not necessarily work. Instead, factory-based instantiation "
```

Comment:
Maybe adjust the existing message in L#383 instead. Add that "reporters may no longer support it" and we're good to go.

Reply:
Fair enough. Initially, I preferred having this mentioned in the error message. But considering that the log warning is printed right before the stacktrace (which I wasn't considering initially when checking the Jira issue's related ML thread post), your proposal should be good enough.

I updated the log message as suggested and force-pushed the change.